flask/flask-common/src/main/java/net/corda/flask/common/LockFile.java

```diff
@@ -24,21 +25,22 @@ private static RandomAccessFile createLockFile(Path path) throws IOException {
     @SneakyThrows
     public static LockFile acquire(Path path, boolean shared) {
         RandomAccessFile raf = createLockFile(path);
-        return new LockFile(raf.getChannel().lock(0L, Long.MAX_VALUE, shared));
+        return new LockFile(raf, raf.getChannel().lock(0L, Long.MAX_VALUE, shared));
```

Comment:
While `RandomAccessFile` is indeed `Closeable`, it is implemented around an instance of `FileDescriptor` which is passed to the `FileChannel` and likely also used by the `FileLock` class. (On UNIX, anyway.)